examples/pytorch/data-cache/fine-tune-with-cache.ipynb

```diff
@@ -0,0 +1,406 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "96fd1dc0-63ea-4eb1-919a-14257eca20d8",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def train_func():\n",
+    "    import time\n",
+    "    import datetime\n",
+    "    from multiprocessing import current_process\n",
+    "    import torch.distributed as dist\n",
+    "    import pyarrow as pa\n",
+    "    import pyarrow.flight\n",
+    "    from pyarrow._flight import FlightClient\n",
+    "    import torch\n",
+    "    import os\n",
+    "    from torch.utils.data import IterableDataset\n",
+    "    from transformers import DistilBertForSequenceClassification, DistilBertTokenizer, AdamW\n",
+    "\n",
+    "    \n",
+    "    # Define DataCacheDataset to access data from cache\n",
+    "    class DataCacheDataset(IterableDataset):\n",
+    "        def __init__(self, seed=0, batch_size=128):\n",
+    "            train_job_name = os.getenv(\"TRAIN_JOB_NAME\")\n",
+    "            self.endpoint = f'grpc://{train_job_name}-cache-service:50051'\n",
+    "            self.seed = seed\n",
+    "            self.batch_size = batch_size\n",
+    "\n",
+    "        def set_epoch(self, epoch):\n",
+    "            self.seed = self.seed + epoch\n",
+    "\n",
+    "        def from_arrow_rb_to_tensor(self, chunk):\n",
+    "            return\n",
+    "\n",
+    "        # Extract RecordBatches and split them into chunks\n",
+    "        def stream_recordbatch_chunks(self, recordbatch_stream, chunk_size):\n",
+    "            leftover = None\n",
+    "        \n",
+    "            for rb in recordbatch_stream:\n",
+    "                batch = rb.data\n",
+    "                if leftover is not None:\n",
+    "                    arrays = [\n",
+    "                        pa.concat_arrays([leftover.column(i), batch.column(i)])\n",
+    "                        for i in range(len(batch.columns))\n",
+    "                    ]\n",
+    "                    batch = pa.RecordBatch.from_arrays(arrays, batch.schema.names)\n",
+    "\n",
+    "                for i in range(0, batch.num_rows, chunk_size):\n",
+    "                    chunk = batch.slice(i, chunk_size)\n",
+    "        \n",
+    "                    if chunk.num_rows < chunk_size:\n",
+    "                        leftover = chunk\n",
+    "                        break\n",
+    "                    else:\n",
+    "                        yield from self.from_arrow_rb_to_tensor(chunk)\n",
+    "                else:\n",
+    "                    leftover = None\n",
+    "        \n",
+    "            if leftover is not None and leftover.num_rows > 0:\n",
+    "                yield from self.from_arrow_rb_to_tensor(leftover)\n",
+    "\n",
+    "        def __iter__(self):\n",
+    "            worker_info = torch.utils.data.get_worker_info()\n",
+    "            if worker_info is None:\n",
+    "                worker_id = 0\n",
+    "                num_workers = 1\n",
```

Comment:
Do we need this ?
When we run PyTorch script with `torchrun`, will get_worker_info always return info ?